cosmos/client.go

```diff
@@ -129,3 +131,34 @@ func (c *Client) BuildAndBroadcastMsg(msg sdktypes.Msg, accName, accPassword str
 		return nil, errors.New("i/o timeout")
 	}
 }
+
+// Stream subscribes to the provided query and returns the hash of the matching ressources.
+func (c *Client) Stream(query string) (chan hash.Hash, error) {
+	subscriber := xstrings.RandAsciiLetters(8)
+	eventStream, err := c.Subscribe(context.Background(), subscriber, query)
+	if err != nil {
+		return nil, err
+	}
+
+	hashC := make(chan hash.Hash)
+	go func() {
+		for event := range eventStream {
+			ressHashes := event.Events[EventHashType]
+			if len(ressHashes) != 1 {
+				// or panic(err) - grpc api do not support
+				// return the errors on the stream for now
+				// so besieds logging the error, it not
+				// much we can do here. same belove
+				continue
+			}
+
+			hash, err := hash.Decode(ressHashes[0])
+			if err != nil {
+				continue
+			}
+			hashC <- hash
+		}
+		c.Unsubscribe(context.Background(), subscriber, query)
```

Comment:
this function will never be called as the eventStream will never be closed...
Except if the eventStream chan is closed..
I think it much more common to return a close or cancel function that the caller have to call.